pkg/oc/cli/newapp/newapp.go

```diff
@@ -443,68 +444,79 @@ func (o *AppOptions) RunNewApp() error {
 		return nil
 	}
 
+	supportedTypes := map[schema.GroupVersionKind]bool{
+		schema.GroupVersionKind{Version: "v1", Kind: "Pod"}:                                    true,
+		schema.GroupVersionKind{Group: buildapi.GroupName, Version: "v1", Kind: "BuildConfig"}: true,
+		schema.GroupVersionKind{Group: imageapi.GroupName, Version: "v1", Kind: "ImageStream"}: true,
+		schema.GroupVersionKind{Group: routeapi.GroupName, Version: "v1", Kind: "Route"}:       true,
+	}
+
 	hasMissingRepo := false
 	installing := []*corev1.Pod{}
 	indent := o.Action.DefaultIndent()
 	containsRoute := false
 	for _, item := range result.List.Items {
 		// these are all unstructured
 		unstructuredObj := item.(*unstructured.Unstructured)
-		obj, err := legacyscheme.Scheme.New(unstructuredObj.GroupVersionKind())
-		if err != nil {
-			return err
-		}
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.Object, obj); err != nil {
-			return err
-		}
-		switch t := obj.(type) {
-		case *corev1.Pod:
-			if t.Annotations[newcmd.GeneratedForJob] == "true" {
-				installing = append(installing, t)
+
+		// Determine if dealing with a "known" resource, containing a switch case below.
+		// If so, go through with a conversion attempt, and fail if necessary.
+		if _, ok := supportedTypes[unstructuredObj.GroupVersionKind()]; ok {
```

Comment:
can you invert this and use a continue?

Also, given the bool, you don't need the `ok`

Reply:
> can you invert this and use a continue?

I think we'd still want to hit the code below this switch case
cc @bparees